src/core/qgsproviderregistry.h

```diff
@@ -107,6 +107,14 @@ class CORE_EXPORT QgsProviderRegistry
      */
     int providerCapabilities( const QString &providerKey ) const;
 
+    /**
+     * Returns the components (e.g. file path, layer name) of a provider uri
```

Comment:
I'd just make a note here that it may not be supported by all providers, and that an empty map will be returned if the provider doesn't support this.